test/server/script-processor-test.js

```diff
@@ -356,115 +356,177 @@ describe('Script processor', function () {
+            }
+            ,
+            {
+                src:      'eval.call(window, "script")',
+                expected: '(eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : eval).call(window, "script")'
+            },
+            {
+                src:      'window.eval.call(window, script)',
+                expected: '(window.eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : window.eval).call(window, script)'
+            },
             {
                 src:      'window["eval"].call(window, script)',
-                expected: 'window["eval"].call(window, __proc$Script(script))'
+                expected: '(window["eval"] === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : window["eval"]).call(window, script)'
             },
 
-            { src: 'eval.apply(window, [script])', expected: 'eval.apply(window, __proc$Script([script], true))' },
-            { src: 'eval.apply(window, ["script"])', expected: 'eval.apply(window, __proc$Script(["script"], true))' },
-            { src: 'eval.apply(window, args)', expected: 'eval.apply(window, __proc$Script(args, true))' },
+            {
+                src:      'eval.apply(window, [script])',
+                expected: '(eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : eval).apply(window, [script])'
+            },
+            {
+                src:      'eval.apply(window, ["script"])',
+                expected: '(eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : eval).apply(window, ["script"])'
+            },
+            {
+                src:      'eval.apply(window, args)',
+                expected: '(eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : eval).apply(window, args)'
+            },
             {
                 src:      'window.eval.apply(window, [script])',
-                expected: 'window.eval.apply(window, __proc$Script([script], true))'
+                expected: '(window.eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : window.eval).apply(window, [script])'
             },
             {
                 src:      'window["eval"].apply(window, [script])',
-                expected: 'window["eval"].apply(window, __proc$Script([script], true))'
+                expected: '(window["eval"] === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : window["eval"]).apply(window, [script])'
+            },
+            {
+                src:      'var tmp = eval;',
+                expected: 'var tmp = eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : eval;'
```

Comment:
Let's use `INTERNAL_LITERAL.tempVar` instead of `script`. Since `script` is too generic and we may shadow `script` var from outer scope for the `eval`. BTW, we should change literal value from `__set$temp` to `__temp$var` for consistency.